src/reynir_correct/pattern.py

```diff
@@ -581,9 +819,9 @@ def wrong_preposition_gagn_að(self, match: SimpleTree) -> None:
     def wrong_preposition_frettir_að(self, match: SimpleTree) -> None:
         """ Handle a match of a suspect preposition pattern """
         # Find the offending preposition
-        pp = match.first_match("P > { 'að' }", self.ctx_að)
+        pp = match.first_match("P > { 'að' }")
```

Comment:
`pp = match.first_match("(P | ADVP) > { 'að' }")` might work fine here, eliminating the need for two checks.